### PR TITLE
feat: Add Async Stack Tagging API support

### PR DESCRIPTION
**Pull request by JiaLiPassion**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

Chrome has an experimental API to improve the debug experience of the
async tasks.
The details can be found here https://bugs.chromium.org/p/chromium/issues/detail?id=332624#c29

This commit add the `async stack tagging` support in the `zone.js`.
User can `import 'zone.js/plugins/async-stack-tagging';` to enable this
feature.

Here is a sample when enable this feature 

image

So we can use `console.trace` to print the async tasks chain information.

### CODE REVIEW

**Review by mgechev — APPROVED** [review]

File: packages/zone.js/test/zone-spec/async-tagging-console.spec.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ifEnvSupports, ifEnvSupportsWithDone} from '../test-util';
+
+fdescribe('AsyncTaggingConsoleTest', () => {
```
Comment (atscott) [inline packages/zone.js/test/zone-spec/async-tagging-console.spec.ts:11]:
```suggestion
describe('AsyncTaggingConsoleTest', () => {
```

**Review by atscott — APPROVED** [review]

**Review by devversion — APPROVED** [review]

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: size-tracking

**Comment by ArielGueta** [comment]
@JiaLiPassion what is different from the current behavior? I don't have this feature and I still see the same as your image:
Screen Shot 2022-07-12 at 9 14 10

See the async tag for setTimeout.

Thanks.

**Comment by JiaLiPassion** [comment]
@ArielGueta , good point, I will ask the author of the `async tagging api` about the difference and let you know the result.

**Comment by JiaLiPassion** [comment]
Thank you @victorporof , here are some screenshots of the `debugger breakpoints`. 

1. without `async tagging` support.
without-async-tagging-debugger

2. with `async tagging` support
with-async-tagging-debugger

**Comment by ArielGueta** [comment]
Thanks @victorporof @JiaLiPassion

**Comment by LanderBeeuwsaert** [comment]
Hi @JiaLiPassion,
is there somewhere a description of how to add this feature for now?
Do we add it in the angular.json scripts array as "node_modules/zone.js/plugins/async-stack-tagging.min"
or do we add it to app module or app component?
On top of that, each user for now needs to add the flag when booting chrome?

Will this be simplified later on and it's better to wait for now or do we go ahead and implement it in our app as-is?

**Comment by JiaLiPassion** [comment]
@LanderBeeuwsaert , to make it work for now, you need to 

1. update `polyfills.ts` 

```
import 'zone.js';
import 'zone.js/plgins/async-stack-tagging';
```

2. in `main.ts`, 
```
const AsyncStackTaggingZoneSpec = (Zone as any)['AsyncStackTaggingZoneSpec'];
Zone.current
  .fork(new AsyncStackTaggingZoneSpec('angular'))
  .run(() => {
    platformBrowserDynamic()
      .bootstrapModule(AppModule)
      .catch((err) => console.error(err));
  });
```

After the new release of the `@angular/core`, you just need to do the `1st` step to make it work, or maybe the `angular cli` will provide some option to do the `1st` step for you.

**Comment by victorporof** [comment]
Heads-up: we are making some changes to the Chrome DevTools console API that drives this feature, so this might not do anything in very recent versions of Chrome Canary. This will be updated in https://github.com/angular/angular/pull/46958.

**Pull request closed without merging**